engines/sword1/control.cpp

```diff
@@ -323,22 +341,24 @@ uint8 Control::runPanel() {
 	_font = (uint8 *)_resMan->openFetchRes(fontId);
 	_redFont = (uint8 *)_resMan->openFetchRes(redFontId);
 
-	uint8 *pal = (uint8 *)_resMan->openFetchRes(SR_PALETTE);
-	uint8 *palOut = (uint8 *)malloc(256 * 3);
-	for (uint16 cnt = 1; cnt < 256; cnt++) {
-		palOut[cnt * 3 + 0] = pal[cnt * 3 + 0] << 2;
-		palOut[cnt * 3 + 1] = pal[cnt * 3 + 1] << 2;
-		palOut[cnt * 3 + 2] = pal[cnt * 3 + 2] << 2;
-	}
-	palOut[0] = palOut[1] = palOut[2] = 0;
-	_resMan->resClose(SR_PALETTE);
-	_system->getPaletteManager()->setPalette(palOut, 0, 256);
-	free(palOut);
+	// Set up mouse
+	_mouse->controlPanel(true);
+
 	uint8 mode = 0, newMode = BUTTON_MAIN_PANEL;
 	bool fullRefresh = false;
-	_mouse->controlPanel(true);
 	uint8 retVal = CONTROL_NOTHING_DONE;
-	_music->startMusic(61, 1);
+
+	// Set up the image and the correct palette for fading up
+	if (SwordEngine::isPsx())
+		destroyButtons();
```

Comment:
This is done to prevent a read access violation within the PSX version. As mentioned in the PR description, the control panel code is the next thing to be rewritten to match the source code (which appears to be the same with minor differences throughout PC, Mac and PSX), because right now it's very different and I had to hack it around to make fades work in there. 

This goes with a promise that there won't be such strange calls in the code.